### PR TITLE
fix #78416: Tempo text oddities in score with linked parts until save/reopen

### PR DESCRIPTION
**Pull request by hpfmn**
always update tempo on tempotextchange and use _text instead of _layout

### CODE REVIEW

**Comment by hpfmn** [comment]
Some more explanation: the if statement just checked whether the tempotext object itself changed - but it didn't check if the tempo is a different one in the song! 
Maybe it is better to have two conditions? like: 
* one for status of tempotext object
* one for status of songtempo

Change from plaintext to xmltext is because plaintext uses _layout which doesn't get updated if the text gets set by the score itself (by method Text::setXmlText) - _layout just gets updated if someone types things into the text
Subpart's texts get updated automatically by updating the text in the main score - but this doesn't update _layout which contains than old data (most likely the default of 80) - which than changes the tempo to wrong values

**Comment by Jojo-Schmitz** [comment]
Rebase needed

File: libmscore/tempotext.cpp [hunk 1]
```diff
@@ -172,21 +172,19 @@ void TempoText::textChanged()
...
                   if (sl.size() == 2) {
                         qreal nt = qreal(sl[1].toDouble()) * pa.f;
-                        if (nt != _tempo) {
-                              setTempo(qreal(sl[1].toDouble()) * pa.f);
-                              if(segment())
-                                    score()->setTempo(segment(), _tempo);
-                              score()->fixTicks();
-                              score()->setPlaylistDirty();
-                              }
+                        setTempo(nt);
```
Comment (Jojo-Schmitz) [inline libmscore/tempotext.cpp]: we could also just drop the local var nt althogether, as it is used only once.

**Comment by Jojo-Schmitz** [comment]
won't this PR collide with #2591 ?

**Comment by hpfmn** [comment]
yes it will collide - but which one gets merged first - I'll just edit the other one ;)

**Comment by IsaacWeiss** [comment]
@lasconic, I may be not the only one to think that this was long since merged, but it seems it wasn't. Would you care to check on it?

**Comment by Jojo-Schmitz** [comment]
Rebase needed

**Comment by Jojo-Schmitz** [comment]
Not only a rebase is needed, it also doesn't seem to work (taking that code and resolving the merge conflicts)

**Comment by anatoly-os** [comment]
The issue is seemed to be fixed by recent changes in tempo mapping.

**Pull request closed without merging**